x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -46,6 +48,8 @@ public interface Role {
 
     Role EMPTY = builder(new RestrictedIndices(Automatons.EMPTY)).build();
 
+    Role EMPTY_RESTRICTED_BY_WORKFLOW = builder(new RestrictedIndices(Automatons.EMPTY)).build();
```

Comment:
I like this better as well. The only catch is that in `WorkflowsRestriction` constructor we have a special handling for an empty case and `null` workflow check since immutable collections do not allow null values when calling `contains`. So I think, it should look like this:

```
...
if (names == null) {
    // No restriction, all workflows are allowed
    this.predicate = name -> true;
} else if (names.isEmpty()) {
    // Empty restriction, no workflow is allowed
    this.predicate = name -> false;
} else {
    this.predicate = name -> {
        if (name == null) {
            return false;
        } else {
            return names.contains(name);
        }
    };
}
```

Reply:
Cool that looks good to me. 

>  immutable collections do not allow null values when calling contains

Right. StringMatcher actually uses HashSet internally. Thanks!